### DISCUSSION TITLE
Second Device Unique ID (duid) on Netbox

### DISCUSSION THREAD

**Discussion by schniefi** [netbox-community/netbox#14052] [Q&A]
I would like to reserve an IPv6 address in a multi-boot system for both the first DUID (Linux) and the second DUID (Windows).
Is it possible to specify a second DUID in one IPv6 reservation? So as in the following example:

`{
  "command": "reservation-add",
  "service": [
    "dhcp6"
  ],
  "arguments": {
    "reservation": {
      "subnet-id": 123,
      "hostname": "xyz",
      "ip-addresses": ["1999:123:56a:1800::123"],
      "duid": "11:11:11:11:11:11",
      "duid2": "22:22:22:22:22:22"
    }
  }
}`

When searching for the IPv6 reservation, both the duid and duid2 should be possible inputs:

`{
  "command": "reservation-get",
  "service": [ "dhcp6" ],
  "arguments": {
    "subnet-id": 123,
    "identifier-type": "duid",
    "identifier": "11:11:11:11:11:11"
  }
}`
or
`{
  "command": "reservation-get",
  "service": [ "dhcp6" ],
  "arguments": {
    "subnet-id": 123,
    "identifier-type": "duid2",
    "identifier": "22:22:22:22:22:22"
  }
}`

For differentiation, the IPv6 reservations should adhere to the following uniqueness criteria:
- unique("ip-address","subnet-id","duid")
- unique("ip-address","subnet-id","duid2")

However, custom fields are not considered by the DHCP6 server for uniqueness checking.

**Comment by candlerb** [comment]
Netbox doesn't have any DUID field in ipam.ip_address, although you could add one as a custom field. If you want two DUIDs on the same IPv6 address then you could have two fields "duid" and "duid2", or a single field with a comma-separated list, or write a plugin where the DUID is a separate model with a link to IPAddress.

Uniqueness constraints can be handled via [custom validators](https://docs.netbox.dev/en/stable/customization/custom-validation/), although there will be race conditions that make it possible to insert two equal values if they arrive very close to each other (but unlikely to happen in practice). With a plugin you can do proper database-backed integrity constraints.

**Comment by schniefi** [comment]
Apologies, I confused the Netbox project with the KEA project. However, it's not necessarily a bad thing that you now have the background of my request. We want to use Netbox alongside KEA for DHCP6. It would be great if Netbox had a DUID and DUID2 in virtualization.models.VMInterface and dcim.models.Interface by default. These would be included out of the box, and it would check if there is already an IPv6 address in the same VLAN with the same DUID or DUID2, similar to how DHCP4 checks if there is an IPv4 address in the same VLAN with the same MAC. While implementing this as custom fields in our system is theoretically possible, I can imagine that more users would consider using Netbox if the out-of-the-box support for DHCP6 were better.
